js/lib/cachetasks.js

```diff
@@ -78,39 +80,41 @@
      *                              otherwise, it passes the request to the
      *                              network via fetch.
      */
-    respondFromCache: async(function*(request, cacheName, strategy = "") {
-      var cache = yield memoizedCaches.open(cacheName);
-      var url = request.url || request;
-      var response = yield cache.match(request);
-      var msg = "";
-      if (response) {
-        return response;
-      }
-      switch (strategy) {
-      case "throw":
-        msg = `Not found in ${cacheName} cache: ${url}`;
-        var error = new Error(msg);
-        console.warn(error);
-        throw error;
-      case "store":
-        try {
+    respondFromCache(request, cacheName, strategy = "") {
+      return async.task(function*() {
+        var cache = yield MemoizedCaches.open(cacheName);
+        var url = request.url || request;
+        var response = yield cache.match(request);
+        var msg = "";
+        if (response) {
+          return response;
+        }
+        switch (strategy) {
+        case "throw":
+          msg = `Not found in ${cacheName} cache: ${url}`;
+          var error = new Error(msg);
+          console.warn(error);
+          throw error;
+        case "store":
+          try {
+            response = yield fetch(request);
+            yield this.putCacheEntry(request, response, cacheName);
+          } catch (err) {
+            var msg = `failed to store ${url} in ${cacheName}.`;
+            console.warn(msg, err);
+            throw err;
+          }
+          break;
+          //Default passes the request to network using fetch()
+        default:
```

Comment:
BTW, all this strategy stuff is going away. Strategy is controlled by HTTP cache directives.